packages/contractkit/src/identity/offchain/storage-writers.ts

```diff
@@ -30,9 +30,9 @@ export class LocalStorageWriter extends StorageWriter {
 }
 
 export class GitStorageWriter extends LocalStorageWriter {
-  async write(data: string, dataPath: string): Promise<void> {
+  async write(data: Buffer, dataPath: string): Promise<void> {
     await this.writeToFs(data, dataPath)
-    execSync(`git add ${dataPath.substr(1)}`, {
+    execSync(`git add ${dataPath}`, {
```

Comment:
Using `exec` here and below is a massive footgun, potentially leading to remote execution, in case we ever path untrusted input to these functions. We should use `spawn` instead (without `shell` mode) https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options